core/parentchain/indirect-calls-executor/src/indirect_calls_executor.rs

```diff
@@ -74,41 +86,25 @@ where
 		self.stf_executor.execute_shield_funds(account, *amount, shard)?;
 		Ok(())
 	}
-
-	fn decrypt_unchecked_extrinsic(
-		&self,
-		xt: UncheckedExtrinsicV4<CallWorkerFn>,
-	) -> Result<(TrustedCallSigned, ShardIdentifier)> {
-		let (call, request) = xt.function;
-		let (shard, cyphertext) = (request.shard, request.cyphertext);
-		debug!("Found CallWorker extrinsic in block: \nCall: {:?} \nRequest: \nshard: {}\ncyphertext: {:?}",
-        	call, bs58::encode(shard.encode()).into_string(), cyphertext);
-
-		debug!("decrypt the call");
-		let shielding_key = self.shielding_key_repo.retrieve_key()?;
-		let request_vec = shielding_key.decrypt(&cyphertext)?;
-
-		Ok(TrustedCallSigned::decode(&mut request_vec.as_slice()).map(|call| (call, shard))?)
-	}
 }
 
-impl<ShieldingKeyRepository, StfExecutor> ExecuteIndirectCalls
-	for IndirectCallsExecutor<ShieldingKeyRepository, StfExecutor>
+impl<ShieldingKeyRepository, StfExecutor, TopPoolAuthor> ExecuteIndirectCalls
+	for IndirectCallsExecutor<ShieldingKeyRepository, StfExecutor, TopPoolAuthor>
 where
 	ShieldingKeyRepository: AccessKey,
 	<ShieldingKeyRepository as AccessKey>::KeyType:
 		ShieldingCryptoDecrypt<Error = itp_sgx_crypto::Error>,
-	StfExecutor: StfExecuteTrustedCall + StfExecuteShieldFunds,
+	StfExecutor: StfExecuteShieldFunds,
+	TopPoolAuthor: AuthorApi<H256, H256> + Send + Sync + 'static,
 {
 	fn execute_indirect_calls_in_extrinsics<ParentchainBlock>(
 		&self,
 		block: &ParentchainBlock,
-	) -> Result<(Vec<OpaqueCall>, Vec<H256>)>
+	) -> Result<Vec<H256>>
 	where
 		ParentchainBlock: ParentchainBlockTrait<Hash = H256>,
 	{
 		debug!("Scanning block {:?} for relevant xt", block.header().number());
-		let mut opaque_calls = Vec::<OpaqueCall>::new();
 		let mut executed_shielding_calls = Vec::<H256>::new();
 		for xt_opaque in block.extrinsics().iter() {
 			// Found ShieldFunds extrinsic in block.
```

Comment:
`handle_shield_funds_xt` effectively also executes indirect calls. Just a little more specific. Shouldn't you adapt this part as well?

Reply:
If that is possible from a business logic point of view, then yes, that would totally make sense. In order to keep the scope of this issue and PR smaller, I kept the shielding call separately still (part of the `Does NOT include` in #711). I will create an issue that I can work on in the next sprint.